executor/executor.cc

```diff
@@ -113,11 +113,24 @@ static void reply_handshake();
 #endif
 
 #if SYZ_EXECUTOR_USES_SHMEM
-const int kMaxOutput = 16 << 20;
+// Allocating (and forking) virtual memory for each executed process is expensive, so we only mmap
+// the amount we might possibly need for the specific received prog.
+const int kMaxOutputComparisons = 14 << 20; // < 1% of all executions.
+const int kMaxOutputCoverage = 6 << 20; // up to 1/3 of all executions (depending on corpus rotation).
+const int kMaxOutputSignal = 4 << 20;
+#ifndef SYZ_EXECUTOR_USES_FORK_SERVER
+const int kMaxOutputNoFork = kMaxOutputComparisons;
```

Comment:
Why not name the const the same way in both modes? Something like `kInitialOutput`. Then we don't need to differentiate between modes when we mmap.